### ISSUE TITLE
Cannot close application unless saving.

### ISSUE THREAD

**Issue by chriseaton** [typora/typora-issues#2180]
On (CentOS) linux, everything seems to work fine, but when attempting to close the application, the "Discard" button does nothing. You have to either save or force-kill the application (xkill). 

![image](https://user-images.githubusercontent.com/5247568/52747654-3360e280-2f99-11e9-88bc-ccf07ac216d2.png)

This is using the *tar.gz x64 version.
![image](https://user-images.githubusercontent.com/5247568/52747681-44a9ef00-2f99-11e9-962c-0730c6bb1d04.png)

**Comment by abnerlee** [comment]
Possible to send us (<hi@typora.io>) the log in https://support.typora.io/Report-Bugs/#logs ?

**Comment by chriseaton** [comment]
I watched the log file and this is what get's generated when I click the "Discard" button (and nothing happens in the UI).

```
ERROR 2/14/2019, 2:42:23 PM  [RenderProcess 1][Error] Error: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /home/person/Downloads/Typora-linux-x64/resources/app/node_modules/pathwatcher/build/Release/pathwatcher.node)
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:166:20)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:740:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:166:20)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at Module.require (file:///home/person/Downloads/Typora-linux-x64/resources/app//window.html:257:25)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/home/person/Downloads/Typora-linux-x64/resources/app/node_modules.asar/pathwatcher/lib/main.js:4:13)
    at Object.<anonymous> (/home/person/Downloads/Typora-linux-x64/resources/app/node_modules.asar/pathwatcher/lib/main.js:266:4)
    at Object.<anonymous> (/home/person/Downloads/Typora-linux-x64/resources/app/node_modules.asar/pathwatcher/lib/main.js:268:3)
    at Module._compile (internal/modules/cjs/loader.js:711:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:722:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at Module.require (file:///home/person/Downloads/Typora-linux-x64/resources/app//window.html:257:25)
    at require (internal/modules/cjs/helpers.js:20:18)
    at n.end (file:///home/person/Downloads/Typora-linux-x64/resources/app//app/window/frame.js:1:389874)
    at file:///home/person/Downloads/Typora-linux-x64/resources/app//app/window/frame.js:1:1305500
    at file:///home/person/Downloads/Typora-linux-x64/resources/app//app/window/frame.js:1:1305570
    at p.callback [as callback_] (file:///home/person/Downloads/Typora-linux-x64/resources/app//app/window/frame.js:1:1040623)
    at HTMLButtonElement.<anonymous> (file:///home/person/Downloads/Typora-linux-x64/resources/app//app/window/frame.js:1:1050689)
    at HTMLDivElement.dispatch (file:///home/person/Downloads/Typora-linux-x64/resources/app//app/window/frame.js:1:1113953)
    at HTMLDivElement.g.handle (file:///home/person/Downloads/Typora-linux-x64/resources/app//app/window/frame.js:1:1111983)
```

**Comment by chriseaton** [comment]
Diving into it, it looks like CXXABI 1.3.9 is only found in a newer GCC version 5.1+ but CentOS/RedHat 7 are using 4.8.5. Not sure of a good workaround.

**Comment by abnerlee** [comment]
Can refer https://support.typora.io/Typora-on-Linux/#trouble-shooting

**Comment by chriseaton** [comment]
Unfortunately the instructions there are for debian-based systems. As mentioned I did some digging, it seems that the libc version supported in CentOS/RHEL 7 is slightly older than what the Typora package is compiled with, likely causing the issue. (seems to be tied to the `pathwatcher` npm package used by Typora). I tried various things found on the net but the concensus is to either build from source yourself (is Typora open-source? I don't see any source code...) or to get a version compiled at the libc compatibility level (basically compiled on a CentOS/RHEL7 machine).
